apps/cow-fi/hooks/useLazyLoadImages.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react'
+
+const LAZY_LOADING_CONFIG = {
+  rootMargin: '10px',
+  placeholderColor: '#f0f0f0',
+  fadeInDuration: '0.3s',
+  minHeight: '100px',
+}
+
+export function useLazyLoadImages() {
+  const replaceImageUrls = React.useCallback((html: string): string => {
+    return html.replace(/<img([^>]*)src="([^"]*)"([^>]*)>/g, (match, before, src, after) => {
+      return `<img${before}data-src="${src}" src="data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='100%' height='100%' viewBox='0 0 1 1' preserveAspectRatio='none'%3E%3Crect width='1' height='1' fill='${LAZY_LOADING_CONFIG.placeholderColor.replace('#', '%23')}' /%3E%3C/svg%3E"${after}>`
```

Comment:
This part is almost all duplicated with line `50`.
Can you extract this into a fn outside of the hook?

Reply:
Addressed this now!